plugin/evm/orderbook_test.go

```diff
@@ -0,0 +1,262 @@
+		BaseAssetQuantity: big.NewInt(0).Mul(size, _1e18),
+		Price:             big.NewInt(0).Mul(price, _1e6),
+		Salt:              salt,
+	}
+	data, err := orderBookABI.Pack("placeOrder", order, []byte("some data"))
+	if err != nil {
+		t.Fatalf("orderBookABI.Pack failed: %v", err)
+	}
+	tx := types.NewTransaction(nonce, orderBookAddress, big.NewInt(0), 8000000, big.NewInt(500000000000), data)
+	signer := types.NewLondonSigner(vm.chainConfig.ChainID)
+	signedTx, err := types.SignTx(tx, signer, privateKey)
+	if err != nil {
+		t.Fatalf("types.SignTx failed: %v", err)
+	}
+	errs := vm.txPool.AddRemotesSync([]*types.Transaction{signedTx})
+	for _, err := range errs {
+		if err != nil {
+			t.Fatalf("lop.txPool.AddOrderBookTx failed: %v", err)
+		}
+	}
+	return signedTx.Hash()
+}
+
+//	  A
+//	 / \
+//	B   C
+//	    |
+//	    D (matching tx of order 1 and 2)
+
+// vm1 proposes block A containing order 1
+// block A is accepted
+// vm1 proposes block B containing order 2
+// vm1 and vm2 set preference to block B
+// vm2 proposes block C containing order 2 & order 3
+// vm1 and vm2 set preference to block C
+// vm2 proposes block D containing matching tx of order 1 and 2
+// vm1 and vm2 set preference to block D
+// reorg happens when vm1 accepts block D
+func TestHubbleLogs(t *testing.T) {
+	// Create two VMs which will agree on block A and then
+	// build the two distinct preferred chains above
+	ctx := context.Background()
+	issuer1, vm1, _, _ := GenesisVM(t, true, genesisJSON, "{\"pruning-enabled\":true}", "")
+	issuer2, vm2, _, _ := GenesisVM(t, true, genesisJSON, "{\"pruning-enabled\":true}", "")
+
+	defer func() {
+		if err := vm1.Shutdown(ctx); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := vm2.Shutdown(ctx); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	// long and short order
+	createPlaceOrderTx(t, vm1, alice, aliceKey, big.NewInt(5), big.NewInt(10), big.NewInt(101))
+	<-issuer1
+	// include alice's long order
+	blocksA := buildBlockAndSetPreference(t, vm1, vm2) // block A - both vms accept
+	accept(t, blocksA...)
+
+	createPlaceOrderTx(t, vm1, bob, bobKey, big.NewInt(-5), big.NewInt(10), big.NewInt(102))
+	<-issuer1
+	// bob's short order
+	buildBlockAndSetPreference(t, vm1) // block B - vm1 only
+
+	// build block C parallel to block B
+	createPlaceOrderTx(t, vm2, bob, bobKey, big.NewInt(-5), big.NewInt(10), big.NewInt(102))
+	createPlaceOrderTx(t, vm2, alice, aliceKey, big.NewInt(5), big.NewInt(11), big.NewInt(104))
+	<-issuer2
+	vm2BlockC := buildBlockAndSetPreference(t, vm2)[0] // block C - vm2 only for now
+
+	vm1BlockC := parseBlock(t, vm1, vm2BlockC)
+	setPreference(t, vm1BlockC, vm1)
+	accept(t, vm1BlockC)
+	accept(t, vm2BlockC)
+
+	detail1 := vm1.limitOrderProcesser.GetOrderBookAPI().GetDetailedOrderBookData(context.Background())
+	detail2 := vm2.limitOrderProcesser.GetOrderBookAPI().GetDetailedOrderBookData(context.Background())
+	t.Logf("VM1 Orders: %+v", detail1)
+	t.Logf("VM2 Orders: %+v", detail2)
```

Comment:
shouldn't we have an assertion on `detail1` and `detail2`?
What assertions fail when @fdgod09's last change is commented out?

Reply:
Asserting that order 2(bob's short order) exists in memory for both VMs